eva-server/src/main/java/uk/ac/ebi/eva/server/ws/contigalias/ContigAliasService.java

```diff
@@ -157,12 +204,76 @@ private String translateContigNameToInsdc(String contigName, String assembly, Co
         return ContigAliasTranslator.getTranslatedContig(contigAliasResponse, ContigNamingConvention.INSDC);
     }
 
+    public List<ContigAliasChromosome> searchChromosomeByName(String contigName, String assembly, ContigNamingConvention contigNamingConvention) {
+        String url = contigAliasUrl + CONTIG_ALIAS_CHROMOSOMES_SEARCH_ENDPOINT + contigName;
+        if ((assembly != null && !assembly.isEmpty()) && contigNamingConvention != null) {
+            url += "?assemblyAccession=" + assembly + "&namingConvention=" + getNameParam(contigNamingConvention);
+        } else if (assembly != null && !assembly.isEmpty()) {
+            url += "?assemblyAccession=" + assembly;
+        } else if (contigNamingConvention != null) {
+            url += "?namingConvention=" + getNameParam(contigNamingConvention);
+        }
+
+        ContigAliasResponse contigAliasResponse = restTemplate.getForObject(url, ContigAliasResponse.class);
+        if (contigAliasResponse == null || contigAliasResponse.getEmbedded() == null ||
+                contigAliasResponse.getEmbedded().getContigAliasChromosomes() == null ||
+                contigAliasResponse.getEmbedded().getContigAliasChromosomes().isEmpty()) {
+            return null;
+        }
+
+        return contigAliasResponse.getEmbedded().getContigAliasChromosomes();
+    }
+
+    public ContigAliasChromosome getUniqueInsdcChromosomeByName(String contigName, String assembly, ContigNamingConvention contigNamingConvention) {
+        List<ContigAliasChromosome> chromosomeList = searchChromosomeByName(contigName, assembly, contigNamingConvention);
+        if (chromosomeList == null || chromosomeList.isEmpty()) {
+            return null;
+        }
+        if (chromosomeList.size() == 1) {
+            return chromosomeList.get(0);
+        } else {
+            Set<String> insdcAccessionsSet = chromosomeList.stream().map(cac -> cac.getInsdcAccession())
+                    .collect(Collectors.toSet());
+            if (insdcAccessionsSet.size() == 1) {
+                return chromosomeList.get(0);
+            } else {
+                Set<ContigNamingConvention> contigNamingConventionSet = chromosomeList.stream()
+                        .map(contigAliasChromosome -> getMatchingContigNamingConvention(contigAliasChromosome, contigName))
+                        .collect(Collectors.toSet());
+                throw new RuntimeException("Multiple Chromosomes found for " + contigName + " in assembly " + assembly
+                        + " with contig naming conventions " + contigNamingConventionSet);
+            }
+        }
+    }
+
+    public ContigNamingConvention getMatchingContigNamingConvention(ContigAliasChromosome contigAliasChromosome, String chromosomeName) {
```

Comment:
This function establish a preference when the same contig name is used by multiple naming convention. 
This is fine but I think I would prefer: INSDC, REFSEQ, ENA_SEQUENCE_NAME, GENBANK_SEQUENCE_NAME, UCSC for the order.

Reply:
Done